Consolidation and FAQ (#838)
commerce-docs/microsite-commerce-storefront: src/content/docs/troubleshooting/faq.mdx

**For end users:** Clear the browser session by closing and reopening the browser, or clear the browsing data to see configuration changes.

## My page isn't showing up

> If a route returns an error or blank content, check whether Edge Delivery Services could load the underlying document from the content pipeline.

Reviewer: ```suggestion
If a route returns an error or empty response, verify that Edge Delivery Services can successfully retrieve the corresponding document from the content pipeline.
```

<Steps>

> 1. Open your browser developer tools and go to the **Network** tab.

Reviewer: ```suggestion
1. Open your browser's developer tools and go to the **Network** tab.
```

> 1. Reload the page and select the **document** request for the URL you are troubleshooting (usually the first HTML request).

Reviewer: ```suggestion
1. Reload the page, then click the main document request (usually the first HTML request).
```

> 1. Open the response **Headers** and look for **`x-error`**. The header value may provide an important signal about missing content.

Reviewer: ```suggestion
In the request details, open **Headers** and look for the `x-error` field. The value can help identify missing content or other issues.
```

</Steps>

> When **`x-error`** reports a failure to load a path ending in **`.md`** from **content-bus** (for example, a `404` for a file under `/products/...`), the page is not available in the Edge Delivery content pipeline. Typical causes include a missing or unpublished document, a wrong path, or content that has not been synced yet.

Reviewer: ```suggestion
If the `x-error` header shows a failure when loading an `.md` file from content-bus (for example, a `404` error under `/products/...`), it means the page is not available in the Edge Delivery content system. This usually happens because the document is missing, not published, the path is incorrect, or the content has not been synced yet.
```

![x-error header on the document response showing a content-bus 404 for a .md path](@images/troubleshoot/eds-x-error-header.png)

> **What to do:** Add or publish the corresponding document in your content space (da.live, for example), or use an **overlay** so EDS can serve HTML for that route. See [BYOM](https://www.aem.live/developer/byom) for more information.

Reviewer: ```suggestion
**What to do:** Add or publish the document in your content system (for example, da.live). If needed, you can also use an overlay so EDS can serve HTML for that route. For more details, see [BYOM](https://www.aem.live/developer/byom).
```

## My product pages return 404

> Product detail pages (PDPs) can return **404** for several reasons. Most likely, the document does not exist in the content pipeline. Check the following:

Reviewer: ```suggestion
Product detail pages (PDPs) may return a 404 when the requested URL cannot be resolved to a document in the Edge Delivery content pipeline. In most cases, this indicates that no corresponding content resource exists or is accessible for that path. Verify the following:
```

> - **Overlay / BYOM not set up:** If your project relies on prerender or bring-your-own-markup (BYOM)patterns so PDPs exist at the edge, confirm that configuration is complete. See [AEM Commerce prerender](/setup/configuration/aem-prerender/).

Reviewer: ```suggestion
**Overlay / BYOM not set up:** If your project depends on prerendering or BYOM (bring your own markup) to serve PDPs, make sure it is fully configured. See [AEM Commerce prerender](/setup/configuration/aem-prerender/).
```

> - **Folder mapping and uppercase SKUs:** When you use the **deprecated** [folder mapping](https://www.aem.live/developer/folder-mapping) approach for PDPs, Edge Delivery Services expects **lowercase** paths. If your catalog uses **uppercase characters in SKUs**, you must supply **metadata** for those URLs (for example via [bulk metadata](https://www.aem.live/docs/bulk-metadata)) so the correct document and metadata are associated with each path. See [SEO metadata](/setup/seo/metadata/#generate-metadata), including the **SKU casing** and **bulk metadata** sections.

Reviewer: ```suggestion
- **Folder mapping and uppercase SKUs:** If you are using the **deprecated** folder mapping method for PDPs, URLs must be lowercase. If your SKUs contain uppercase letters, you need to provide metadata (for example, using bulk metadata) so the correct content is matched to each URL. See:
- [Folder mapping](https://www.aem.live/developer/folder-mapping)
- [Bulk metadata](https://www.aem.live/docs/bulk-metadata)
- [SEO metadata](/setup/seo/metadata/#generate-metadata)
```
